Cakefile

```diff
@@ -437,21 +461,25 @@ runTests = (CoffeeScript) ->
       CoffeeScript.run code.toString(), {filename, literate}
     catch error
       failures.push {filename, error}
-  return !failures.length
+
+  Promise.all asyncTests
```

Comment:
I'd suggest trying to get a 'final' pass/fail from this method if possible, e.g. something like:

```coffee
Promise.all(asyncTests).then  ->
  Promise.reject() if failures.length
```

Then, when handling the results, we can just do:

```coffee
runTests CoffeeScript
  .catch -> process.exit 1
```

Reply:
Good call!